### PR TITLE
Upgrade deps

### PR DESCRIPTION
**Pull request by SimenB**
Noticed greenkeeper complaining all over, so went for a bump

### CODE REVIEW

File: jsconfig.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
```
Comment (SimenB) [inline jsconfig.json]: what's this file for?

File: src/repository.ts [hunk 2]
```diff
@@ -5,10 +5,9 @@ import { FeatureInterface } from './feature';
 import { resolve } from 'url';
 import { get } from './request';
 import { CustomHeaders } from './unleash';
+import { Response } from 'request';
 
-export interface StorageImpl {
-    new (Storage);
-}
+export type StorageImpl = typeof Storage;
```
Comment (SimenB) [inline src/repository.ts]: I think this is correct?

File: test/unleash.network.test.js [hunk 3]
```diff
@@ -31,5 +31,5 @@ test.cb('should emit network errors', t => {
         process.nextTick(() => {
             t.end();
         });
-    }, 0);
+    }, 5);
```
Comment (SimenB) [inline test/unleash.network.test.js]: Had to bump this, or `destroy` was called before the `fetch` to `client/features` was completed, which meant the error handler was removed, which means an unhandled error bubbled up to node, which causes ava to fail the test (good change!)

File: jsconfig.json (hunk 1 above)
Comment (ivarconr) [reply jsconfig.json]: I have no idea.

File: src/client.ts [hunk 4]
```diff
@@ -31,16 +31,12 @@ export default class UnleashClient extends EventEmitter {
...
-            }
-            return false;
-        });
-        return match;
+    private getStrategy(name: string): Strategy | undefined {
+        return this.strategies.find(
+            (strategy: Strategy): boolean => {
+                return strategy.name === name;
+            },
+        );
```
Comment (ivarconr) [inline src/client.ts]: :+1:

File: src/metrics.ts [hunk 5]
```diff
@@ -20,20 +20,20 @@ export interface MetricsOptions {
 interface Bucket {
     start: Date;
     stop: Date | null;
-    toggles: Object;
+    toggles: { [s: string]: { yes: number; no: number } };
```
Comment (ivarconr) [inline src/metrics.ts]: :+1:

File: test/client.test.js [hunk 6]
```diff
@@ -103,21 +103,6 @@ test('should use custom strategy', t => {
     t.true(result);
 });
 
-test('should use a set of custom strategies', t => {
```
Comment (ivarconr) [inline test/client.test.js]: Why is this test not needed?

**Comment by ivarconr** [comment]
Nice job! I would like to verify the client via a few manual tests also before merging this.

File: test/client.test.js (hunk 6 above)
Comment (SimenB) [reply test/client.test.js]: Duplicate. Ava throws on tests with same title (why I changed the title of some others)

File: src/unleash.ts [hunk 7]
```diff
@@ -1,8 +1,8 @@
 import Client from './client';
 import Repository from './repository';
 import Metrics from './metrics';
-import { Strategy, defaultStrategies } from './strategy/index';
-export { Strategy } from './strategy/index';
+import { Strategy, defaultStrategies } from './strategy';
```
Comment (ivarconr) [inline src/unleash.ts]: This change has broken defaultStrategies.

Example:
```
➜  unleash-client-node git:(upgrade-deps) ✗ node examples/simple_usage.js
/home/ivaosthu/code/unleash/unleash-client-node/lib/unleash.js:67
        strategies = strategy_1.defaultStrategies.concat(strategies);
                                                  ^

TypeError: Cannot read property 'concat' of undefined
    at new Unleash (/home/ivaosthu/code/unleash/unleash-client-node/lib/unleash.js:67:51)
    at initialize (/home/ivaosthu/code/unleash/unleash-client-node/lib/index.js:10:16)
    at Object.<anonymous> (/home/ivaosthu/code/unleash/unleash-client-node/examples/simple_usage.js:3:16)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Function.Module.runMain (module.js:694:10)
    at startup (bootstrap_node.js:204:16)
```

Comment (SimenB) [reply src/unleash.ts]: Works for me?

```sh-session
$ node -v
v10.14.1
$ node examples/simple_usage.js
Fetching toggles from: http://unleash.herokuapp.com
featureX enabled: false
featureX enabled: false
featureX enabled: false
featureX enabled: false
featureX enabled: false
featureX enabled: false
featureX enabled: false
featureX enabled: false
featureX enabled: false
featureX enabled: false
featureX enabled: false
featureX enabled: false
```

Are you able to add a test that fails?

Comment (SimenB) [reply src/unleash.ts]: Diff in that file from master:

```diff
diff --git i/lib/unleash.js w/lib/unleash.js
index 1f83fdf..25a6f87 100644
--- i/lib2/unleash.js
+++ w/lib2/unleash.js
@@ -1,21 +1,23 @@
 "use strict";
 var __extends = (this && this.__extends) || (function () {
-    var extendStatics = Object.setPrototypeOf ||
-        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
-        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
+    var extendStatics = function (d, b) {
+        extendStatics = Object.setPrototypeOf ||
+            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
+            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
+        return extendStatics(d, b);
+    };
     return function (d, b) {
         extendStatics(d, b);
         function __() { this.constructor = d; }
         d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
     };
 })();
-exports.__esModule = true;
+Object.defineProperty(exports, "__esModule", { value: true });
 var client_1 = require("./client");
 var repository_1 = require("./repository");
 var metrics_1 = require("./metrics");
-var index_1 = require("./strategy/index");
-var index_2 = require("./strategy/index");
-exports.Strategy = index_2.Strategy;
+var strategy_1 = require("./strategy");
+exports.Strategy = strategy_1.Strategy;
 var os_1 = require("os");
 var events_1 = require("events");
 var os_2 = require("os");
@@ -48,24 +50,23 @@ var Unleash = /** @class */ (function (_super) {
             }
             catch (e) {
                 //unable to read info;
-                info = {};
             }
-            var prefix = info.username
+            var prefix = info
                 ? info.username
                 : "generated-" + Math.round(Math.random() * 1000000) + "-" + process.pid;
             instanceId = prefix + "-" + os_2.hostname();
         }
-        _this.repository = new repository_1["default"]({
+        _this.repository = new repository_1.default({
             backupPath: backupPath,
             url: url,
             appName: appName,
             instanceId: instanceId,
             refreshInterval: refreshInterval,
-            headers: customHeaders
+            headers: customHeaders,
         });
-        strategies = index_1.defaultStrategies.concat(strategies);
+        strategies = strategy_1.defaultStrategies.concat(strategies);
         _this.repository.on('ready', function () {
-            _this.client = new client_1["default"](_this.repository, strategies);
+            _this.client = new client_1.default(_this.repository, strategies);
             _this.client.on('error', function (err) { return _this.emit('error', err); });
             _this.client.on('warn', function (msg) { return _this.emit('warn', msg); });
             _this.emit('ready');
@@ -77,14 +78,14 @@ var Unleash = /** @class */ (function (_super) {
         _this.repository.on('warn', function (msg) {
             _this.emit('warn', msg);
         });
-        _this.metrics = new metrics_1["default"]({
+        _this.metrics = new metrics_1.default({
             disableMetrics: disableMetrics,
             appName: appName,
             instanceId: instanceId,
             strategies: strategies.map(function (strategy) { return strategy.name; }),
             metricsInterval: metricsInterval,
             url: url,
-            headers: customHeaders
+            headers: customHeaders,
         });
         _this.metrics.on('error', function (err) {
             err.message = "Unleash Metrics error: " + err.message;
```

Comment (ivarconr) [reply src/unleash.ts]: You are absolutely right. My problem was that `lib/` is not cleaned after a build and I had an additional `strategy.js` file directly in that folder in addiotion to the `lib/strategy`.

Would it make sense for build to first clean the lib folder? What is the common practice with TypeScript?

**Pull request merged**